Gems/Atom/RPI/Code/Source/RPI.Public/View.cpp

```diff
@@ -47,17 +47,20 @@ namespace AZ
         {
             AZ_Assert(!name.IsEmpty(), "invalid name");
 
-            // Set default matrixes. 
+            // Set default matrices 
             SetWorldToViewMatrix(AZ::Matrix4x4::CreateIdentity());
             AZ::Matrix4x4 viewToClipMatrix;
             AZ::MakePerspectiveFovMatrixRH(viewToClipMatrix, AZ::Constants::HalfPi, 1, 0.1f, 1000.f, true);
             SetViewToClipMatrix(viewToClipMatrix);
 
-            Data::Asset<ShaderAsset> viewSrgShaderAsset = RPISystemInterface::Get()->GetCommonShaderAssetForSrgs();
-
-            if (viewSrgShaderAsset.IsReady())
+            if (auto rpiSystemInterface = RPISystemInterface::Get())
```

Comment:
This could be an issue because if we allow view created successfully without ViewSRG get created, the view won't be work for rendering. 

Maybe we can add creating ViewSRG later in UpdateSrg() when the view is used.

Reply:
If this condition would only happen in test, maybe we could add comment for it.

Reply:
If the `RPISystemInterface` needs to be there, then I think the test should create a mock `RPISystemInterface`. That way the production code can continue unmodified, but the test can continue to test just what it needs.

Reply:
The problem with adding a mock is that interface is pretty huge, I mean I could do it but I'd rather factor the code to not need this whole interface to get its job done, it would be better to pass in the required dependency and have the interface be as small as possible. I _can_ add the mock but it's a load of busy work and doesn't improve the design at all (why I avoided it in this case to begin with).

Reply:
@VickyAtAZ and I discussed offline and have an update to resolve things

Reply:
https://github.com/o3de/o3de/pull/5494/commits/cc0474753e13fe7f976d7b9d82873523f7e948f7

Reply:
Getting a failure in `RenderPipelineTests, ViewFunctionsTest` with the `OnAddToRenderPipeline()` addition - it seems like it's legit for `m_shaderResourceGroup` to be null in those tests and the `AZ::Error` is causing a problem.

See here - https://jenkins.build.o3de.org/job/O3DE/view/change-requests/job/PR-5494/4/console

Reply:
The reason it fails is because the `viewSrgShaderAsset` is not ready...

Reply:
As a temporary workaround as I'd really like to get this PR merged today, I've downgraded the Error to a Warning. Is this something you could then look into offline please @VickyAtAZ (https://github.com/o3de/o3de/pull/5494/commits/2c02e65437c3f44f76d6e8e3a91aaa0a52cdc3f1)